### PR TITLE
finish search Venue

### PR DESCRIPTION
**Pull request by at-phutran**
- Search Venue
- UI Search

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: ATFoody/View/Controller/Search/SearchViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//  Created by Phu Tran on 11/15/17.
+//  Copyright © 2017 Asian Tech. All rights reserved.
+//
+
+import MVVM
+import GoogleMaps
+import RealmS
+import RealmSwift
+
+class SearchViewModel: ViewModel {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/SearchViewModel.swift]: Missing `final` keyword. Please keep in mind that. Define `final class` if you would like its not inherited.

File: ATFoody/View/Controller/Search/SearchViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+// MARK: - Call API
+extension SearchViewModel {
+    enum SearchCompletion {
+        case success
+        case failure(Error)
+    }
+
+    func getSearchVenues(near: String, query: String, completion: @escaping (SearchCompletion) -> Void) {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/SearchViewModel.swift]: Rename `func searchVenues()`

File: ATFoody/View/Controller/Search/SearchViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+extension SearchViewModel {
+    func numberOfItems(inSection section: Int) -> Int {
+        return trendings.count
+    }
+
+    func numberOfSections() -> Int {
+        return 1
+    }
+
+    func viewModelForItem(at indexPath: IndexPath) -> ViewModel {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/SearchViewModel.swift]: Should be returned specifically view model.

File: ATFoody/View/Controller/Search/TableViewSearch/SearchListViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+//  Created by Phu Tran on 11/17/17.
+//  Copyright © 2017 Asian Tech. All rights reserved.
+//
+
+import Foundation
+import SwiftUtils
+import UIKit
+import MVVM
+
+final class SearchListViewModel: ViewModel {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/TableViewSearch/SearchListViewModel.swift]: Please rename this class

File: ATFoody/View/Controller/Search/TableViewSearch/SearchListViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    var name: String?
+    var address = ""
+    var thumnailPath: String?
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/TableViewSearch/SearchListViewModel.swift]: Check misspellings

File: ATFoody/View/Controller/Search/TableViewSearch/SearchTableViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+//  SearchTableViewCell.swift
+//  ATFoody
+//
+//  Created by Phu Tran on 11/16/17.
+//  Copyright © 2017 Asian Tech. All rights reserved.
+//
+
+import UIKit
+
+class SearchTableViewCell: UITableViewCell {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/TableViewSearch/SearchTableViewCell.swift]: Missing `final` keyword. Should be rename SearchCell

File: ATFoody/View/Controller/Search/TableViewSearch/SearchTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    @IBOutlet weak var thumbnailImage: UIImageView!
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var addressLabel: UILabel!
+    @IBOutlet weak var ratingLabel: UILabel!
+
+    var viewModel: SearchListViewModel? {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/TableViewSearch/SearchTableViewCell.swift]: ViewModel is none `Optional`

File: ATFoody/View/Controller/Search/TableViewSearch/SearchTableViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        didSet {
+            guard let viewModel = viewModel else { return }
+            nameLabel.text = viewModel.name
+            addressLabel.text = viewModel.address
+            ratingLabel.text = String(viewModel.rating)
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/TableViewSearch/SearchTableViewCell.swift]: Handle `type data` in view model

File: ATFoody/Model/API/Services/API.Search.swift [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+                case .success(let value):
+                    guard let json = value as? JSObject,
+                        let response = json["response"] as? JSObject,
+                        let groups = response["groups"] as? JSArray,
+                        let group = groups.first,
+                        let items = group["items"] as? JSArray else {
+                            completion(.failure(Api.Error.json))
+                            return
+                    }
+                    var trendings: [Trending] = []
```
Comment (at-quangphu) [inline ATFoody/Model/API/Services/API.Search.swift]: No need create `trendings `

File: ATFoody/View/Controller/Search/SearchViewController.swift [hunk 10]
```diff
@@ -7,11 +7,78 @@
 //
 
 import UIKit
+import GoogleMaps
+import RealmS
+import SwiftUtils
 
-final class SearchViewController: UIViewController {
+class SearchViewController: UIViewController {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/SearchViewController.swift]: Missing `final` keyword

File: ATFoody/View/Controller/Search/SearchViewController.swift [hunk 11]
```diff
@@ -7,11 +7,78 @@
...
+    var viewModel = SearchViewModel()
+    // MARK: - Life circle
     override func viewDidLoad() {
         super.viewDidLoad()
+        title = App.String.search
+        searchTableView.register(SearchCell.self)
+        searchTableView.dataSource = self
+    }
+
+    @IBAction func cancelQuery(_ sender: Any) {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/SearchViewController.swift]: Move these `@IBAction func` into MARK: - IBAction

File: ATFoody/View/Controller/Search/SearchViewController.swift [hunk 12]
```diff
@@ -7,11 +7,78 @@
...
+
+// MARK: - UITableViewDataSoure
+extension SearchViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.numberOfItems(inSection: section)
+    }
 
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeue(SearchCell.self)
+        if let vm = viewModel.viewModelForItem(at: indexPath) as? SearchCellViewModel {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/SearchViewController.swift]: `viewModel.viewModelForItem(at: indexPath)` should be returned a specific type.

File: ATFoody/Model/API/Services/API.Search.swift [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+            return ["ll": ll,
+                "near": near,
+                "query": query,
+                "venuePhotos": venuePhotos,
+                "client_id": clientId,
+                "client_secret": clientSecret,
+                "v": v]
+        }
+
+        init(ll: String, near: String, query: String, venuePhotos: Int) {
```
Comment (at-quangphu) [inline ATFoody/Model/API/Services/API.Search.swift]: No need `venuePhotos ` property. Its constant

File: ATFoody/View/Controller/Search/TableViewSearch/SearchCell.swift [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+//  ATFoody
+//
+//  Created by Phu Tran on 11/16/17.
+//  Copyright © 2017 Asian Tech. All rights reserved.
+//
+
+import UIKit
+
+final class SearchCell: UITableViewCell {
+    @IBOutlet weak var thumbnailImage: UIImageView!
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Search/TableViewSearch/SearchCell.swift]: Rename `thumbnailImageView`

**Pull request merged**